cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/partitioned/ConsumerConfiguration.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.dataset.lib.partitioned;
+
+import co.cask.cdap.api.Predicate;
+import co.cask.cdap.api.dataset.lib.PartitionDetail;
+
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
+
+/**
+ * Configuration parameters to be used by a {@link PartitionConsumer}.
+ */
+public class ConsumerConfiguration {
+  /**
+   * Instance of ConsumerConfiguration with default parameters
+   */
+  public static final ConsumerConfiguration DEFAULT = builder().build();
+
+  private final Predicate<PartitionDetail> partitionPredicate;
```

Comment:
In the current design, this predicate (replacement for PartitionFilter) is used when populating the working set (from the partitioned file set).
There's also a `PartitionAcceptor` that's used when the client queries the partition consumer (pulls from the working set), which has a method that returns an enum.